### ISSUE TITLE
train_UDE Issue when loss_function = "marginal likelihood"

### ISSUE THREAD

**Issue by b-l-proctor** [zechmeunier/emude#33]
I ran into an error for two of the ten `train_UDE` combinations. Both occurred when the `loss_function` was set to "marginal likelihood" 

`train_UDE(NODE1_norm, loss_function = "marginal likelihood", optimizer = "BFGS"`
<img width="1008" height="459" alt="Image" src="https://github.com/user-attachments/assets/85fc3aa6-eb02-4d95-bd2d-eb8c01a541f4" />

`train_UDE(NODE1_norm, loss_function = "marginal likelihood", optimizer = "ADAM"`
<img width="1013" height="459" alt="Image" src="https://github.com/user-attachments/assets/41c2f5fd-bb7b-43f0-b404-18fda810e477" />

**Comment by zechmeunier** [comment]
This is consistent with how the Julia function `train!` is constructed when given a marginal likelihood loss function.

```
 elseif loss_function == "marginal likelihood"

    L = size(UDE.data)[1]
    Pν = errors_to_matrix(0.1, L)
    if :process_error in keys(loss_options)
      Pν = errors_to_matrix(loss_options.process_error, L)
    end

    Pη = 0
    if :observation_error in keys(loss_options)
      Pη = errors_to_matrix(loss_options.observation_error, L)
    else
      throw("Marginal likelihood requires observation errors.")
    end
```

We could set the default value of observation error (Pη) to something small like we do for process error (Pν). Or we can just write better help text to tell the R users that they have to specific an observation error in the `loss.options` kwarg.

@Jack-H-Buckner What would you recommend here?
